webapp/src/components/github_repo_selector.jsx

```diff
@@ -4,130 +4,67 @@
 import React, {PureComponent} from 'react';
 import PropTypes from 'prop-types';
 
-import debounce from 'debounce-promise';
-import AsyncSelect from 'react-select/async';
-
-import {getStyleForReactSelect} from 'utils/styles';
-
 import Client from 'client';
+import ReactSelectSetting from 'components/react_select_setting';
 
-const searchDebounceDelay = 400;
+const initialState = {
+    invalid: false,
+    error: null,
+    repoOptions: [],
+};
 
 export default class GithubRepoSelector extends PureComponent {
     static propTypes = {
         required: PropTypes.bool,
         theme: PropTypes.object.isRequired,
         onChange: PropTypes.func.isRequired,
-        value: PropTypes.object,
-        showErrors: PropTypes.bool,
+        value: PropTypes.string,
+        addValidate: PropTypes.func,
+        removeValidate: PropTypes.func,
     };
 
     constructor(props) {
         super(props);
-
-        this.state = {
-            invalid: true,
-            error: null,
-        };
+        this.state = initialState;
     }
 
-    componentDidUpdate(prevProps, prevState) {
-        // check prevProps against current props value to ensure validationError
-        // doesn't show on initial component rendering
-        if (prevState.invalid && this.props.value && (this.props.value.full_name !== prevProps.value.full_name)) {
-            this.setState({invalid: false}); //eslint-disable-line react/no-did-update-set-state
-        }
+    componentDidMount() {
+        this.getRepos();
     }
 
-    handleGetRepoValue = () => {
-        return this.debouncedSearchRepo();
-    };
-
-    searchRepo = () => {
+    getRepos = () => {
         return Client.getRepositories().then((data) => {
```

Comment:
Can we move `getRepositories` into an action instead of directly accessing the `Client` in the component? This would require you to put this component in a folder, then `connect` it with an `index.js` file, like the [TimeSidebar](https://github.com/mattermost/mattermost-plugin-github/tree/master/webapp/src/components/team_sidebar) component. Then pass in the action as a prop using `mapDispatchToProps`.

The action won't end up storing the result in the redux store, but I think the code would be cleaner using props, as opposed to accessing `Client` directly.